crates/transpiler/src/passes/basis_translator/mod.rs

```diff
@@ -47,80 +51,92 @@ use crate::equivalence::EquivalenceLibrary;
 use crate::target::Qargs;
 use crate::target::QargsRef;
 use crate::target::Target;
-use crate::TranspilerError;
 
-type InstMap = IndexMap<GateIdentifier, BasisTransformOut, ahash::RandomState>;
-type ExtraInstructionMap<'a> = IndexMap<&'a PhysicalQargs, InstMap, ahash::RandomState>;
+type AhashIndexMap<K, V> = IndexMap<K, V, ahash::RandomState>;
+type AhashIndexSet<O> = IndexSet<O, ahash::RandomState>;
+type InstMap = AhashIndexMap<GateIdentifier, BasisTransformOut>;
+type ExtraInstructionMap<'a> = AhashIndexMap<&'a PhysicalQargs, InstMap>;
 type PhysicalQargs = SmallVec<[PhysicalQubit; 2]>;
 
-#[allow(clippy::too_many_arguments)]
-#[pyfunction(name = "base_run", signature = (dag, equiv_lib, qargs_with_non_global_operation, min_qubits, target_basis=None, target=None, non_global_operations=None))]
-pub fn run_basis_translator(
-    py: Python<'_>,
-    dag: &DAGCircuit,
+#[pyfunction(name = "base_run", signature = (dag, equiv_lib, min_qubits, target=None, target_basis=None))]
+fn py_run_basis_translator(
+    dag: &mut DAGCircuit,
     equiv_lib: &mut EquivalenceLibrary,
-    qargs_with_non_global_operation: HashMap<Qargs, HashSet<String>>,
     min_qubits: usize,
+    target: Option<&mut Target>,
     target_basis: Option<HashSet<String>>,
-    target: Option<&Target>,
-    non_global_operations: Option<HashSet<String>>,
 ) -> PyResult<Option<DAGCircuit>> {
+    run_basis_translator(dag, equiv_lib, min_qubits, target, target_basis).map_err(|e| e.into())
+}
+
+pub fn run_basis_translator(
+    dag: &mut DAGCircuit,
```

Comment:
This doesn't need to be `mut` you're not mutating the dag anywhere. You have L86 calling `as_deref_mut()` but nothing is actually mutating it.

Reply:
Fixed in cbe4e1d868234c50cce87a47723cbac688aea0e7